src/txn/txn.c

```diff
@@ -2175,6 +2175,7 @@ int
 __wt_txn_rollback_required(WT_SESSION_IMPL *session, const char *reason)
```

Comment:
It looks like this function is only called once in the code, which explains why you are always seeing the same message.

Reply:
I think it's called more than once.

Reply:
I believe it's called a handful of times. The string passed in gets saved into session->txn->rollback_reason, but we don't seem to print this reason out.

Reply:
We print it in `__session_commit_transaction` and `__wt_cache_eviction_worker`.

Reply:
Ah I missed one, it's called twice. But in the second case it is called with "oldest pinned transaction ID rolled back for eviction", which is showing in the log.

Reply:
I think what you want here is to add messaging around scenarios like the one in e.g. src/cursor/cur_file.c:416?

Reply:
I believe we are printing the rollback reason when one is available, but the code does not set a reason every time the `WT_ROLLBACK` error is returned. Note that the code you cited in `__session_commit_transaction` checks if `txn->rollback_reason` is null.

Reply:
We probably shouldn't set `rollback_reason` in any new places. That may break many tests in WT and MDB.

Reply:
This change is already being done by the code, so I'm trying to figure out what you want to achieve so I can help :)